packages/core/src/material/UnlitMaterial.ts

```diff
@@ -86,16 +88,42 @@ export class UnlitMaterial extends Material {
         break;
       case AlphaMode.Blend:
         {
+          this.shaderData.disableMacro("ALPHA_CUTOFF");
+
           target.sourceColorBlendFactor = target.sourceAlphaBlendFactor = BlendFactor.SourceAlpha;
           target.destinationColorBlendFactor = target.destinationAlphaBlendFactor = BlendFactor.OneMinusSourceAlpha;
           target.colorBlendOperation = target.alphaBlendOperation = BlendOperation.Add;
           depthState.writeEnabled = false;
           this.renderQueueType = RenderQueueType.Transparent;
         }
         break;
+      case AlphaMode.CutOff:
+        {
+          this.shaderData.enableMacro("ALPHA_CUTOFF");
+
+          target.sourceColorBlendFactor = target.sourceAlphaBlendFactor = BlendFactor.One;
+          target.destinationColorBlendFactor = target.destinationAlphaBlendFactor = BlendFactor.Zero;
+          target.colorBlendOperation = target.alphaBlendOperation = BlendOperation.Add;
+          depthState.writeEnabled = true;
+          this.renderQueueType = RenderQueueType.AlphaTest;
+        }
+        break;
     }
   }
 
+  /**
+   * Alpha cutoff value.
+   * @remarks fragments with alpha channel lower than cutoff value will be discarded.
+   */
+  get alphaCutoff(): number {
+    return this._alphaCutoff;
+  }
+
+  set alphaCutoff(v: number) {
+    this._alphaCutoff = v;
+    this.shaderData.setFloat("u_alphaCutoff", v);
+  }
+
   /**
    * Whether to render both sides.
```

Comment:
We can use more power interface like `cullFace` or `renderFace`, and developer can set `Back`, `Front`, `None`.